File: backend/src/main/java/com/student/backend/dto/EventTaskReportResponse.java
package com.student.backend.dto;

import com.student.backend.model.EventStatus;

import java.util.List;
import java.util.Map;

public record EventTaskReportResponse(
        String eventId,
        String eventName,
        EventStatus eventStatus,
        Map<String, Long> tasksByStatus,
        List<TaskReportLine> tasks
) {
    public record TaskReportLine(
            String taskId,
            String title,
            String status,
            String zoneName,
            int performerCount
    ) {}
}

File: mobile/app/src/main/java/com/example/mobile/data/model/EventTaskReportResponse.kt
package com.example.mobile.data.model

data class EventTaskReportResponse(
    val eventId: String,
    val eventName: String,
    val eventStatus: String,
    val tasksByStatus: Map<String, Long>,
    val tasks: List<TaskReportLine>
) {
    data class TaskReportLine(
        val taskId: String,
        val title: String,
        val status: String,
        val zoneName: String?,
        val performerCount: Int
    )
}
